conbench/entities/benchmark_result.py

```diff
@@ -223,38 +223,27 @@ def create(userres) -> "BenchmarkResult":
         case = Case.get_or_create({"name": benchmark_name, "tags": tags})
         context = Context.get_or_create({"tags": userres["context"]})
         info = Info.get_or_create({"tags": userres.get("info", {})})
-
-        # Create a corresponding `Run` entity in the database if it doesn't
-        # exist yet. Use the user-given `id` (string) as primary key. If the
-        # Run is already known in the database then only update the
-        # `has_errors` property, if necessary. All other run-specific
-        # properties provided as part of this BenchmarkResultCreate structure
-        # (like `machine_info` and `run_name`) get silently ignored.
-        run = Run.first(id=userres["run_id"])
-        if run:
-            if "error" in userres:
-                run.has_errors = True
-                run.save()
+        if "machine_info" in userres:
+            hardware = Machine.get_or_create(userres["machine_info"])
         else:
-            hardware_info_field = (
-                "machine_info" if "machine_info" in userres else "cluster_info"
-            )
-            Run.create(
-                {
-                    "id": userres["run_id"],
-                    "name": userres.pop("run_name", None),
-                    "reason": userres.pop("run_reason", None),
-                    "github": userres.pop("github", None),
-                    hardware_info_field: userres.pop(hardware_info_field),
-                    "has_errors": "error" in userres,
-                }
+            hardware = Cluster.get_or_create(userres["cluster_info"])
+
+        user_given_commit_info: Optional[TypeCommitInfoGitHub] = userres.get("github")
+        commit = None
+        if user_given_commit_info is not None:
+            commit = commit_fetch_info_and_create_in_db_if_not_exists(
+                user_given_commit_info
             )
-            # The above's `create()` might fail (race condition), in which case
-            # we can re-read, but then we also have to call
-            # `validate_run_result_consistency(userres)` one more time (because
-            # _we_ are not the ones who created the Run).
```

Comment:
It's good to see this type of concern go away. (I mean, why were' doing this :)).